### PR TITLE
Implemented BindableLayout EmptyView

### LINKED ISSUE #6932: [Enhancement] EmptyView for BindableLayout
## Summary

Add an EmptyView property to BindableLayout which, when set, displays if there are no other items in the layout.

This is basically a duplicate of the EmptyView feature on CollectionView.

## API Changes

In `BindableLayout`, add:

`public static readonly BindableProperty EmptyViewProperty =
BindableProperty.Create(nameof(EmptyView), typeof(object), typeof(Layout<View>), null);`

`public static readonly BindableProperty EmptyViewTemplateProperty =
BindableProperty.Create(nameof(EmptyViewTemplate), typeof(DataTemplate), typeof(Layout<View>), null);`

## Intended Use Case

In the following example, if `ItemsSource` contains any items, they will be displayed in the StackLayout and EmptyView will not be visible. If `ItemSource` contains no items, the EmptyView will be displayed as the sole element in the StackLayout.

```
<StackLayout BindableLayout.ItemsSource="{Binding ItemsSource}">
    <BindableLayout.EmptyView>
        <Label Text="No Results"/>
    </BindableLayout.EmptyView>
</StackLayout>
```

If EmptyViewTemplate is set, the template content will be displayed with EmptyView as the binding context for the template content.

If EmptyViewTemplate is _not_ set, the following rules apply:

- If the content of EmptyView is a simple string, that string is displayed as a Label. 
- If the content of EmptyView is a Forms element, that element is displayed. 
- Otherwise, the EmptyView object's `ToString(InvariantCulture)` representation is displayed as a Label.

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change ###

This change adds a property to the `BindableLayout` that lets users define an `EmptyView`. Whenever there are no items in the `BindableLayout.ItemsSource` the `EmptyView` or `EmptyViewTemplate` is shown.

### Issues Resolved ### 

- fixes #6932

### API Changes ###

Added:
 - object BindableLayout.EmptyView { get; set; } //Bindable Property
 - DataTemplate BindableLayout.EmptyViewTemplate {get; set; }  //Bindable Property

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

Nothing happens if the user doesn't start using the new properties. When they do and the `BindableLayout.ItemsSource` is empty, the empty view will show up accordingly with how it is configured through the two properties.

By using the `EmptyView` the user can specify either a simple string that will be shown in a `Label` or define their own `View` that will be shown.

The user could also choose for the `EmptyViewTemplate` to show a `DataTemplate`. The `EmptyViewTemplate.BindingContext` will inherit the `BindingContext` from the `BindableLayout`.

When both the `EmptyView` and `EmptyViewTemplate` are set, the `EmptyViewTemplate` always takes precedence,

### Before/After Screenshots ### 

TBD

### Testing Procedure ###

TBD

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/BindableLayout.cs [hunk 1]
```diff
@@ -173,12 +230,28 @@ void CreateChildren()
...
 				layout.Children.Add(CreateItemView(item, layout));
 			}
 		}
 
+		void UpdateEmptyView(Layout<View> layout)
+		{
+			if (_currentEmptyView == null)
+				return;
+
+			if (!_itemsSource.GetEnumerator().MoveNext())
```
Comment (rmarinho) [inline Xamarin.Forms.Core/BindableLayout.cs]: _itemsSource can be null no?

Comment (jfversluis) [reply Xamarin.Forms.Core/BindableLayout.cs]: Good catch, thanks!

**Comment by jfversluis** [comment]
@jsuarezruiz that looks a bit nicer! Done and a rebase just as a bonus.

**Comment by rmarinho** [comment]
On iOS when i remove i still see the items and  it shows the empty view too.

image

image

**Comment by jfversluis** [comment]
Hm that is strange... Must be something that is changed then, this now also happens in the original BindableLayout gallery page. Let me have a look. Thanks!

**Comment by jfversluis** [comment]
I have rebased and removed the shadow from this gallery page. The above bug is not because of this change.

I have created a separate issue for the "frame shadow"-bug here #8567, that way this PR can move forward

**Pull request merged**